app/javascript/dashboard.js

```diff
@@ -0,0 +1,432 @@
+      Delete
+    </a>`;
+
+  const instance = M.Modal.getInstance(document.getElementById('delete_garden_modal'));
+  instance.open();
+}
+
+function delete_garden_modal_close() {
+  const instance = M.Modal.getInstance(document.getElementById('delete_garden_modal'));
+  instance.close();
+}
+
+export function add_garden_modal() {
+  const instance = M.Modal.getInstance(document.getElementById('add_garden_modal'));
+  instance.open();
+}
+
+function add_garden_modal_close() {
+  const instance = M.Modal.getInstance(document.getElementById('add_garden_modal'));
+  instance.close();
+}
+
+function update_garden_modal(garden_id) {
+  document.getElementById('update_garden_modal_footer').innerHTML = `
+    <a onclick="update_garden_modal_close()" class="modal-close waves-effect waves-green btn-flat">
+      Cancel
+    </a>
+    <a onclick="update_garden(${garden_id})" class="modal-close waves-effect waves-green btn-flat">
+      Update
+    </a>`;
+
+  const instance = M.Modal.getInstance(document.getElementById('update_garden_modal'));
+  instance.open();
+}
+
+export function update_garden_modal_close() {
+  const instance = M.Modal.getInstance(document.getElementById('update_garden_modal'));
+  instance.close();
+}
+
+export function populate_table_garden() {
+  getJSON('./gardens', (data) => {
+    for (let i = 0; i <= data.length - 1; i++) {
+      document.getElementById('garden_table').innerHTML += `
+        <tr>
+          <td>${data[i].name}</td>
+          <td>${data[i].address}</td>
+          <td>${data[i].lat}, ${data[i].long}</td>
+          <td>${data[i].contact_name}</td>
+          <td>${data[i].contact_number}</td>
+          <td>
+            <ul id=garden_id${data[i].id}></ul>
+          </td>
+          <td>${data[i].notes}</td>
+          <td>
+            <a onclick='delete_garden_modal(${data[i].id})' class="waves-effect waves-teal btn-flat">
+              <i class="material-icons">delete</i>
+            </a>
+            <a onclick='update_garden_modal(${data[i].id})' class="waves-effect waves-teal btn-flat">
+              <i class="material-icons">edit</i>
+            </a>
+          </td>
+        </tr>`;
+
+      console.log(typeof data[i].contact_name);
+    }
+    getJSON('./garden_produces', (data) => {
+      for (let i = 0; i <= data.length - 1; i++) {
+        document.getElementById(`garden_id${data[i].garden_id}`).innerHTML += `
+          <li>
+            <span class="produce_id${data[i].produce_id}"></span>,
+            ${data[i].available_at}
+          </li>`;
+      }
+      getJSON('./produces', (data) => {
+        for (let i = 0; i <= data.length - 1; i++) {
+          const elements = document.getElementsByClassName(`produce_id${i}`);
+          for (let j = 0; j < elements.length; j++) {
+            elements[j].innerHTML += data[i].name;
+          }
+        }
+      }, (status) => {
+        alert('Something went wrong.');
```

Comment:
These are really sad alert messages. They are fine for now (as we might transition away from the code which requires them) but a future dev / user might appreciate a more descriptive message.